### PR TITLE
Refactor the client cache to use string file id

### PR DESCRIPTION
**Pull request by bf8086**
This allows certain client filesystem to use UUID as file id without worrying about collision.

### CODE REVIEW

**Comment by bf8086** [comment]
@apc999 @calvinjia PTAL.

File: core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheFileInStream.java [hunk 1]
```diff
@@ -112,12 +112,13 @@ public int read(byte[] b, int off, int len) throws IOException {
...
     int bytesRead = 0;
+    long lengthToRead = Math.min(len, mStatus.getLength() - mPosition);
     // for each page, check if it is available in the cache
-    while (bytesRead < len && mPosition < mStatus.getLength()) {
+    while (bytesRead < lengthToRead) {
       long currentPage = mPosition / mPageSize;
       int currentPageOffset = (int) (mPosition % mPageSize);
-      int bytesLeftInPage = (int) Math.min(mPageSize - currentPageOffset, len - bytesRead);
-      PageId pageId = new PageId(mStatus.getFileId(), currentPage);
+      int bytesLeftInPage = (int) Math.min(mPageSize - currentPageOffset, lengthToRead - bytesRead);
```
Comment (calvinjia) [inline core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheFileInStream.java]: Good catch

File: core/client/fs/src/main/java/alluxio/client/file/cache/store/RocksPageStore.java [hunk 2]
```diff
@@ -151,9 +155,10 @@ public void close() {
   }
 
   private static byte[] getKeyFromPageId(PageId pageId) {
-    ByteBuffer buf = ByteBuffer.allocate(KEY_LEN);
-    buf.putLong(pageId.getFileId());
+    byte[] fileId = pageId.getFileId().getBytes();
+    ByteBuffer buf = ByteBuffer.allocate(Long.BYTES + fileId.length);
     buf.putLong(pageId.getPageIndex());
+    buf.put(fileId);
```
Comment (calvinjia) [inline core/client/fs/src/main/java/alluxio/client/file/cache/store/RocksPageStore.java]: Is it intentional to reverse the key order?

File: core/client/fs/src/main/java/alluxio/client/file/cache/store/RocksPageStore.java [hunk 3]
```diff
@@ -163,12 +168,12 @@ public void close() {
...
   @Nullable
   private static PageId getPageIdFromKey(byte[] key) {
-    if (key.length != KEY_LEN) {
+    if (key.length < Long.BYTES) {
       return null;
     }
     ByteBuffer buf = ByteBuffer.wrap(key);
-    long fileId = buf.getLong();
     long pageIndex = buf.getLong();
+    String fileId = Charset.defaultCharset().decode(buf).toString();
```
Comment (calvinjia) [inline core/client/fs/src/main/java/alluxio/client/file/cache/store/RocksPageStore.java]: Do you know how much more expensive this is compared to `getLong`?

File: core/client/fs/src/main/java/alluxio/client/file/cache/store/RocksPageStore.java (hunk 2 above)
Comment (bf8086) [reply core/client/fs/src/main/java/alluxio/client/file/cache/store/RocksPageStore.java]: yeah, it is easier to read the rest of the buffer as string when we decode the page id from byte array.

File: core/client/fs/src/main/java/alluxio/client/file/cache/store/RocksPageStore.java (hunk 3 above)
Comment (bf8086) [reply core/client/fs/src/main/java/alluxio/client/file/cache/store/RocksPageStore.java]: Probably a lot more expensive. Not sure how much it impact the overall performance though.

Comment (calvinjia) [reply core/client/fs/src/main/java/alluxio/client/file/cache/store/RocksPageStore.java]: yeah something we can keep in mind if performance is an issue

**Comment by bf8086** [comment]
alluxio-bot, test this please.

**Comment by apc999** [comment]
alluxio-bot, merge this please

**Pull request merged**